paddle/fluid/distributed/collective/process_group_nccl.cc

```diff
@@ -860,6 +862,44 @@ void ProcessGroupNCCL::CreateNCCLEnvCache(const Place& place,
   auto comm_ctx = std::make_unique<phi::GPUContext>(place);
   comm_ctx->set_nccl_comm(nccl_comm);
 
+  // gather global ranks in current group
+  int* gpu_global_rank = nullptr;
```

Comment:
1. Use tensor instead of raw data since cuda memory APIs  are not efficient?
2. If use raw data, check the result of CUDA API.

Reply:
> 1. Use tensor instead of raw data since cuda memory APIs  are not efficient?
> 2. If use raw data, check the result of CUDA API.

fixed